vm/src/stdlib/array.rs

```diff
@@ -440,7 +538,23 @@ impl PyArray {
         obj: PyObjectRef,
         vm: &VirtualMachine,
     ) -> PyResult<()> {
-        self.borrow_value_mut().setitem(needle, obj, vm)
+        match needle {
+            Either::A(i) => self.borrow_value_mut().setitem_by_idx(i, obj, vm),
+            Either::B(slice) => {
+                let items = match obj.payload::<PyArray>() {
```

Comment:
If you want to check reference equality you can change `&self` to `zelf: PyRef<Self>`, and then you can do `zelf.is(&obj)` to check if they're the same